internal/manager/config.go

```diff
@@ -130,8 +130,8 @@ func (c *Config) FlagSet() *pflag.FlagSet {
 	// Kubernetes configurations
 	flagSet.StringVar(&c.KubeconfigPath, "kubeconfig", "", "Path to the kubeconfig file.")
 	flagSet.StringVar(&c.IngressClassName, "ingress-class", annotations.DefaultIngressClass, `Name of the ingress class to route through this controller.`)
-	flagSet.BoolVar(&c.EnableLeaderElection, "leader-elect", false, "Enable leader election for controller manager. Enabling this will ensure there is only one active controller manager.")
```

Comment:
Related to the "Do we want the same for 2.1.x?" question above.

IMO we remove this one immediately:
- We do not intend to honor it at all. At most the deprecated flag would print a warning log if not set to the default.
- I expect that usage of this flag in the wild is low: it wasn't present in 1.x and users that upgraded were accustomed to it being handled automatically, and we didn't add documentation indicating you should enable it in DB mode AFAIK. If people _were_ using it, I expect we'd have received a report about it deleting configuration when enabled, and we didn't.
- Upgrade burden is pretty small. If you weren't using it, you do nothing. If you were, you'll see your upgrade fail and print a clear "unknown flag" error, and you run a new upgrade that removes it.

Reply:
Resolving this one and continuing in the CHANGELOG.md thread.

Reply:
> We do not intend to honor it at all. At most the deprecated flag would print a warning log if not set to the default.

True, but we also know that what we set it to is what the end-user would have to have set it to or it would have broken their deployment.

> I expect that usage of this flag in the wild is low: it wasn't present in 1.x and users that upgraded were accustomed to it being handled automatically, and we didn't add documentation indicating you should enable it in DB mode AFAIK. If people were using it, I expect we'd have received a report about it deleting configuration when enabled, and we didn't.
> Upgrade burden is pretty small. If you weren't using it, you do nothing. If you were, you'll see your upgrade fail and print a clear "unknown flag" error, and you run a new upgrade that removes it.

I get where you're coming from, but we have two threads going now so I'm going to try and stick with https://github.com/Kong/kubernetes-ingress-controller/pull/2053#discussion_r769681802 and follow up with you there.